ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -256,9 +249,12 @@ def add_policy(self, name_behavior_id: str, policy: TFPolicy) -> None:
                     self.__class__.__name__
                 )
             )
-        if not isinstance(policy, PPOPolicy):
-            raise RuntimeError("Non-PPOPolicy passed to PPOTrainer.add_policy()")
+        if not isinstance(policy, NNPolicy):
+            raise RuntimeError("Non-NNPolicy passed to PPOTrainer.add_policy()")
         self.policy = policy
+        self.optimizer = PPOOptimizer(self.policy, self.trainer_parameters)
```

Comment:
ppo/trainer.py and sac/trainer.py seem to have a lot in common. Any way to address this ?

Reply:
This will be a future PR :P The intention is to make general trainers that have at their core only how it gets data from the buffer (on-policy vs. off-policy) so that most of the methods can move to RLTrainer.